src/plot_api/plot_api.js

```diff
@@ -384,14 +384,16 @@ Plotly.plot = function(gd, data, layout, config) {
         drawData,
         finalDraw,
         initInteractions,
-        Plots.rehover
+        Plots.rehover,
+        Plots.previousPromises
     ];
 
-    Lib.syncOrAsync(seq, gd);
-
     // even if everything we did was synchronous, return a promise
     // so that the caller doesn't care which route we took
-    return Promise.all(gd._promises).then(function() {
```

Comment:
This was the crux of the issue: if you have more than one async item in `seq`, only the promises added to `gd._promises` by the first one will exist when this `Promise.all` is called (at the synchronous return of `Lib.syncOrAsync`), so any added after that will not be awaited. But `Plots.previousPromises` (added to the end of `seq` above) looks at `gd._promises` when it's actually invoked so it gets them all.

Reply:
Solid investigation work :microscope: :tada: :1st_place_medal: 

Is this enough to close https://github.com/plotly/plotly.js/issues/76 you think?

Reply:
> Is this enough to close #76 you think?

I think so!